modelskill/obs.py

```diff
@@ -116,11 +116,10 @@ def weight(self, value: float) -> None:
     # TODO: move this to TimeSeries?
     @staticmethod
     def _parse_time(time):
-        if not isinstance(time.to_index(), pd.DatetimeIndex):
-            raise TypeError(
-                f"Input must have a datetime index! Provided index was {type(time.to_index())}"
-            )
-        return time.dt.round("100us")
+        if isinstance(time, pd.DatetimeIndex):
```

Comment:
Not directly related to the PR, but necessary for the simplest possible test case.